pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -33,6 +34,17 @@
 # TODO(@tchaton): Typing-pickle issue on python<3.7 (https://github.com/cloudpipe/cloudpickle/pull/318)
 _METRIC = Any  # Union[Metric, torch.Tensor]
 _METRIC_COLLECTION = Union[_METRIC, Mapping[str, _METRIC]]
+_OUT_METRIC = Union[torch.Tensor, Dict[str, torch.Tensor]]
+_PBAR_METRIC = Union[float, Dict[str, float]]
+_OUT_DICT = Dict[str, _OUT_METRIC]
+_PBAR_DICT = Dict[str, _PBAR_METRIC]
+
+
+class _METRICS(TypedDict):
```

Comment:
is this `TypedDict` in all python versions we support?

Reply:
Yes: https://github.com/PyTorchLightning/pytorch-lightning/blob/1e08b044ec7785addf939be94f8fdd286e6bd87f/requirements.txt#L13